File: denco-content-hub-frontend/src/api/transcription.ts
/**
 * Temporary manual API for transcription.
 * TODO: Remove after regenerating hey-api client with transcription endpoints.
 */

import { client } from '@/api/client/client.gen'

export interface TranscriptionSegment {
  start: number
  end: number
  text: string
  speaker?: string | null
}

export interface TranscriptionResponse {
  id: number
  content_item_id: number
  status: string
  text: string | null
  language: string | null
  duration_seconds: number | null
  segments: TranscriptionSegment[] | null
  whisper_model: string | null
  error_message: string | null
  created_at: string
  updated_at: string
}

export async function getTranscriptionApi(
  workspaceId: number,
  contentId: number,
): Promise<TranscriptionResponse> {
  const result = await client.get({
    url: '/api/v1/workspaces/{workspace_id}/content/{content_id}/transcription',
    path: {
      workspace_id: workspaceId,
      content_id: contentId,
    },
    throwOnError: true,
  })
  return result.data as TranscriptionResponse
}

File: denco-content-hub-backend/app/schemas/transcription.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TranscriptionSegment(BaseModel):
    """Single transcription segment with timecodes."""

    start: float
    end: float
    text: str
    speaker: str | None = None


class TranscriptionResponse(BaseModel):
    """Full transcription response."""

    model_config = ConfigDict(from_attributes=True)

    id: int | None = None
    content_item_id: int
    status: str
    text: str | None = None
    language: str | None = None
    duration_seconds: int | None = None
    segments: list[TranscriptionSegment] | None = None
    whisper_model: str | None = None
    error_message: str | None = None
    diarization_status: str | None = None
    diarization_error: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class TranscriptionShortResponse(BaseModel):
    """Short transcription info for content lists."""

    model_config = ConfigDict(from_attributes=True)

    status: str
    language: str | None = None
    duration_seconds: int | None = None
    diarization_status: str | None = None
